internal/pkg/admission/reconciler.go

```diff
@@ -65,7 +65,19 @@ func (r *Reconciler) ReconcileDeletion(
 		errors = append(errors, err)
 	}
 
-	// TODO delete webhooks, configmap and service
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      policyServer.NameWithPrefix(),
+			Namespace: r.DeploymentsNamespace,
+		},
+	}
+	err = r.Client.Delete(ctx, service)
+	if err != nil {
+		r.Log.Error(err, "ReconcileDeletion: cannot delete PolicyServer Service %s", policyServer.Name)
+		errors = append(errors, err)
+	}
```

Comment:
On deletion I think we should be cautious with erroring out on all cases. For example, I would not return if the error is a 404 (`IsNotFound`) of "children" resources. When reconciling deletions we want to delete as much as possible, and if an issue happens at the middle of the deletion reconciliation, but the error is that the resource being deleted is no longer persisted, is fine to keep going to the next step.

Reply:
To sum up: I would not accumulate `IsNotFound` errors and just assume that is good. An extreme case: if the reconciliation finishes and all it did was error on every step because all resources are giving a 404, then we are done reconciling, and that is fine.

Reply:
yes, you are right. I'll check if the error is different that IsNotFound in the next commit. Someone could even manually remove a resource created by the operator, then delete the PolicyServer. In that case the reconciliation would fail because the resource was not there.